test/integration/controllermanager/quota/quota_test.go

```diff
@@ -0,0 +1,157 @@
+	var (
+		providerType string
+		resourceName string
+		objectKey    client.ObjectKey
+
+		secret        *corev1.Secret
+		quota         *gardencorev1beta1.Quota
+		secretbinding *gardencorev1beta1.SecretBinding
+	)
+
+	BeforeEach(func() {
+		providerType = "provider-type"
+		resourceName = "test-" + utils.ComputeSHA256Hex([]byte(CurrentSpecReport().LeafNodeLocation.String()))[:8]
+		objectKey = client.ObjectKey{Namespace: testNamespace.Name, Name: resourceName}
+
+		secret = &corev1.Secret{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+		}
+
+		quota = &gardencorev1beta1.Quota{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+			Spec: gardencorev1beta1.QuotaSpec{
+				Scope: corev1.ObjectReference{
+					APIVersion: "v1",
+					Kind:       "Secret",
+				},
+			},
+		}
+
+		secretbinding = &gardencorev1beta1.SecretBinding{
+			ObjectMeta: kutil.ObjectMetaFromKey(objectKey),
+			Provider: &gardencorev1beta1.SecretBindingProvider{
+				Type: providerType,
+			},
+			SecretRef: corev1.SecretReference{
+				Name:      resourceName,
+				Namespace: testNamespace.Name,
+			},
+			Quotas: []corev1.ObjectReference{
+				{
+					Name:      resourceName,
+					Namespace: testNamespace.Name,
+				},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create Secret")
+		Expect(testClient.Create(ctx, secret)).To(Succeed())
+		log.Info("Created secret for test", "secret", client.ObjectKeyFromObject(secret))
+
+		By("Create Quota")
+		Expect(testClient.Create(ctx, quota)).To(Succeed())
+		log.Info("Created quota for test", "quota", client.ObjectKeyFromObject(quota))
+
+		DeferCleanup(func() {
+			By("Delete Quota")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, quota))).To(Succeed())
+		})
+
+		if secretbinding != nil {
+			By("Create SecretBinding")
+			Expect(testClient.Create(ctx, secretbinding)).To(Succeed())
+			log.Info("Created secretbinding for test", "secretbinding", client.ObjectKeyFromObject(secretbinding))
+
+			DeferCleanup(func() {
+				By("Delete SecretBinding")
+				Expect(client.IgnoreNotFound(gardener.ConfirmDeletion(ctx, testClient, secretbinding))).To(Succeed())
+				Expect(client.IgnoreNotFound(testClient.Delete(ctx, secretbinding))).To(Succeed())
+			})
+		}
+	})
+
+	Context("no secretbinding referencing Quota", func() {
+		BeforeEach(func() {
+			secretbinding = nil
+		})
+
+		It("should add the finalizer and release it on deletion", func() {
+			By("Ensure finalizer got added")
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, objectKey, quota)).To(Succeed())
+				g.Expect(quota.Finalizers).To(ConsistOf("gardener"))
+			}).Should(Succeed())
+
+			By("Delete Quota")
+			Expect(testClient.Delete(ctx, quota)).To(Succeed())
+
+			By("Ensure Quota is released")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(quota), quota)
+			}).Should(BeNotFoundError())
+		})
+	})
+
+	Context("secretbinding referencing Quota", func() {
```

Comment:
```suggestion
	Context("SecretBinding referencing Quota", func() {
```

(same for other occurrences)